freemarker/src/main/java/org/jdbi/v3/freemarker/FreemarkerSqlLocator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.freemarker;
+
+import java.io.File;
+import java.io.FileReader;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import net.jodah.expiringmap.ExpirationPolicy;
+import net.jodah.expiringmap.ExpiringMap;
+
+/**
+ * Locates SQL in <code>.sql.ftl/code> Freemarker files on the classpath.
+ */
+public class FreemarkerSqlLocator {
+    private static final Map<String, Template> CACHE = ExpiringMap.builder()
```

Comment:
why do we evict cache at all? can templates really change in the lifetime of an application? 

I would suggest a static cache instead.

Reply:
Ostensibly we evict to help with class unloading in a JEE Container. In practice I'm not sure it makes a difference.

Reply:
I suggest just leaving it as it is for now. But I found a problem in the project where i am using jdbi. We need a way to toggle the cache on and off. Maybe add an option on the annotation for not using the cache. Its a bit annoying having to restart spring boot for every tiny little change in the template ...

Reply:
that said you could use unit tests for verifying those tiny small changes, but sometimes live debugging is much faster

Reply:
I don't really remember the use case, but some users were generating dynamic templates at runtime.  If there's no expiration mechanism you will eventually run out of heap.

Retaining references to classes causing classloader leaks is a problem as well, but in this case the key seems to be of type `String`.

Reply:
Well leave the cache then. But disable it somehow? Lazy initializing the cache? (synchronized lazy loading). By using a parameter on @UseFreemarkerSqlLocator(cache=false)

Reply:
It is unfortunate that there's no way to configure these caches. I'm not sure an annotation attribute would help since the core class is using a static cache.

I think it would be reasonable to use e.g. environment variables or Java properties to control their configuration.

It might also be interesting to introduce a configuration class and move the caching into there, which would reduce this class to strictly loading templates from the classpath. @stevenschlansker can you envision any backward compatibility problems with this idea? I could see possibly reduced performance for core code, but I don't anticipate anything actually breaking.

Reply:
Ok. I recall having lots of discussions that resulted in the current structure, I just don't remember all the details of those discussions so I might have forgotten some of the reasons we did it.

Reply:
@qualidafial it had nothing to do with freemarker. I just had to understand how it works. ${keys} when using ```@BindBeanList``` is a String in the template model. I thought it was a sequence, but it is made a string to be able to control this outside the template for complex bindings. 

I have added the test  (copy pasted from stringtemplate4 module) that proves it now also works with freemarker.

Reply:
ah wrong thread

Reply:
I suggest closing this discussion. The cache do no harm. So if we can get this thing merged then i will plan to integrate the new module into our production app as soon as possible. An RC candidate would be great. But i dont know where we are in the way towards 3.2.0

Reply:
@qualidafial @jarlah Configuring the cache as you describe is an interesting idea.  And, I disagree that it "does no harm".  We've had users report essentially this exact bug before: https://github.com/jdbi/jdbi/issues/93
It's fine that it got merged but I will file a followup ticket to track this.

Reply:
https://github.com/jdbi/jdbi/issues/1116